src/Selection.js

```diff
@@ -113,92 +191,109 @@ class Selection {
         bottom: offsetData.bottom + bottom,
         right: offsetData.right + right
       },
-      { top: e.pageY, left: e.pageX });
+      { top: pageY, left: pageX });
 
       if (!collides) return;
     }
 
-    let result = this.emit('mousedown', this._mouseDownData = {
-      x: e.pageX,
-      y: e.pageY,
-      clientX: e.clientX,
-      clientY: e.clientY
+    let result = this.emit('beforeSelect', this._initialEventData = {
+      isTouch: /^touch/.test(e.type),
+      x: pageX,
+      y: pageY,
+      clientX,
+      clientY,
     });
 
     if (result === false)
       return;
 
     //e.preventDefault();
```

Comment:
I know this wasn't part of your PR, but this commented line looks fairly clean-up-able.